scripts/docker_setup.sh

```diff
@@ -98,10 +100,61 @@ function move_ifaces() {
 			sudo ip netns exec pause ethtool -N "${ifaces[$i]}" flow-type tcp4 action 0
 			sudo ip netns exec pause ethtool -u "${ifaces[$i]}"
 		fi
+		if [ "$mode" == 'cndp' ]; then
+			# num queues
+			num_q=1
+			# start queue index
+			start_q_idx=22
+			# RSS using TC filter
+			setup_tc "${ifaces[$i]}" $num_q $start_q_idx
+		fi
 	done
 	setup_addrs
 }
 
+# Setup TC
+# Parameters: $1 = interface, $2 = number of queues, $3 = start queue index
+function setup_tc() {
+	# Interface name
+	iface=$1
+	# Number of queues
+	num_q=$2
+	# Start queue index
+	sq_idx=$3
+	sudo ip netns exec pause ethtool --offload $iface hw-tc-offload on
+	# Create two traffic control groups for the two queue sets - set 0 and set 1.
+	# queue set 1 will be used for dataplane traffic.
+	# queue set 0 will handle rest of the traffic (eg: control plane traffic).
+	# for e.g., 22@0 means 22 queues starting from queue id 0.
+	# 4@22 mean 4 queues starting from queue id 22.
+	sudo ip netns exec pause tc qdisc add dev $iface root mqprio \
+		num_tc 2 map 0 1 queues $sq_idx@0 $num_q@$sq_idx hw 1 mode channel
+	sudo ip netns exec pause tc qdisc add dev $iface clsact
+}
+
+# Add TC rules for N3/N6/N9 access and core interface.
+# Parameters: $1 = access interface, $2 = core interface
+# Inner UE IP address range and GTPU port is hardcoded for now.
+# UE IP address should match the generated traffic pattern.
+function add_tc_rules() {
```

Comment:
Should we add a note clarifying/stating that this function is used only for `CNDP`?

Reply:
ok